src/coreclr/inc/profilepriv.inl

```diff
@@ -1712,7 +1715,7 @@ inline BOOL CORProfilerTrackJITInfo()
             (&g_profControlBlock)->globalEventMask.IsEventMaskSet(COR_PRF_MONITOR_JIT_COMPILATION));
```

Comment:
Do we actually need `CORProfilerPresent` check in all these methods? Would it be better to do just the `g_profControlBlock)->globalEventMask.IsEventMaskSet` check? It would reduce the profiler early out to just a single condition like what it was before the notification profilers feature.

Reply:
I think you're right, we don't actually need the CORProfilerPresent, we can just do the globalEventMask check, since the profiler callback code will check if the profilers are active before calling in. As a profiler is detaching we might do a few unnecessary iterations, but that is better than always doing an extra check